cmd/thanos/store.go

```diff
@@ -185,29 +189,35 @@ func runStore(
 		return errors.Wrap(err, "create object storage store")
 	}
 
-	begin := time.Now()
-	level.Debug(logger).Log("msg", "initializing bucket store")
-	if err := bs.InitialSync(context.Background()); err != nil {
-		return errors.Wrap(err, "bucket store initial sync")
-	}
-	level.Debug(logger).Log("msg", "bucket store ready", "init_duration", time.Since(begin).String())
-
-	ctx, cancel := context.WithCancel(context.Background())
-	g.Add(func() error {
-		defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
-
-		err := runutil.Repeat(syncInterval, ctx.Done(), func() error {
-			if err := bs.SyncBlocks(ctx); err != nil {
-				level.Warn(logger).Log("msg", "syncing blocks failed", "err", err)
+	// bucketStoreReady signals when bucket store is ready.
+	bucketStoreReady := make(chan struct{})
+	{
+		begin := time.Now()
```

Comment:
This is weird as it shows also non init bucket operation time.. Wonder if we should not keep it together in run.Group?